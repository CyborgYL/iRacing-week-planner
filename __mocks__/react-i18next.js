import moment from 'moment';
import i18n from 'i18next';

i18n
  .init({
    resources: {},
    debug: false,
    keySeparator: false,
    nsSeparator: false,
    interpolation: {
      escapeValue: false,
      format: (value, format) => {
        if (value instanceof Date) {
          return moment(value).format(format);
        }

        return value;
      },
    },
  });

export function useTranslation() {
  return {
    t: i18n.t.bind(i18n),
  };
}
export function withTranslation() {
  return (Component) => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: i18n.t.bind(i18n) };
    return Component;
  };
}
