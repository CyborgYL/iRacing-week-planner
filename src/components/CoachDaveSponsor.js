// @flow

import * as React from 'react';
import { useTranslation } from 'react-i18next';

import mainStyles from '../styles/main.module.scss';
import styles from './styles/coachDave.module.scss';

import '@blueprintjs/core/lib/css/blueprint.css';

export default function CoachDaveSponsor(): React.Node {
  const { t } = useTranslation();
  const clickSponsor = () => {
    window.dataLayer.push({
      event: 'event',
      eventProps: {
        category: 'Sponsor button',
        action: 'click',
        label: 'Coach Dave',
      },
    });
  };

  return (
    <div className={styles.sponsoredContainer}>
      <a
        onClick={clickSponsor}
        href="https://coachdaveacademy.com/product-category/iracing-setups/?ref=iracing-week-planner.tmo.lol"
        rel="sponsored"
        className={styles.sponsoredLink}
      >
        <h4 className={styles.sponsoredHeader}>
          {t('Sponsored')}
        </h4>
        <h3 className={styles.sponsoredTitle}>
          {t('Looking for iRacing Setups?')}
        </h3>
        <p className={styles.sponsoredBody}>
          {t('Grab a setup from Coach Dave Academy, developed by professional sim racers and engineers.')}
        </p>
        <p
          className={`${mainStyles.btn} ${mainStyles['btn-primary']} ${styles.button}`}
        >
          {t('Browse Setups')}
        </p>
      </a>
    </div>
  );
}
