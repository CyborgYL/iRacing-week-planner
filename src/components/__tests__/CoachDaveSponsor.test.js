import React from 'react';
import { create, act } from 'react-test-renderer';
import { describe, test } from '@jest/globals';
import axios from 'axios';
import { useTranslation } from 'react-i18next';

import CoachDaveSponsor from '../CoachDaveSponsor';

jest.mock('react-i18next');
jest.mock('axios');

describe('components/CoachDaveSponsor', () => {
  test('renders', async () => {
    let component;

    act(() => {
      component = create(<CoachDaveSponsor />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    await act(async () => {
      await component.root.findByType('a').props.onClick();
    });

    expect(window.dataLayer.push).toHaveBeenCalledWith(expect.objectContaining({ event: 'select_content' }));
  });
});
