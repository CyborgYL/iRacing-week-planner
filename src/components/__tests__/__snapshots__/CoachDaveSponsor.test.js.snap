// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CoachDaveSponsor renders 1`] = `
<div
  className="sponsoredContainer"
>
  <a
    className="sponsoredLink"
    href="https://coachdaveacademy.com/product/iracing-setup-subscription/?ref=weekly_planner"
    onClick={[Function]}
    rel="sponsored"
  >
    <h4
      className="sponsoredHeader"
    >
      Sponsored
    </h4>
    <h3
      className="sponsoredTitle"
    >
      Looking for iRacing Setups?
    </h3>
    <p
      className="sponsoredBody"
    >
      Grab a setup from Coach Dave Academy, developed by professional sim racers and engineers.
    </p>
    <p
      className="btn btn-primary button"
    >
      Browse Setups
    </p>
  </a>
</div>
`;
